src/main/java/org/terasology/logic/health/RegenComponent.java

```diff
@@ -22,53 +22,33 @@
 import java.util.Map;
 
 public class RegenComponent implements Component {
-    public long lowestEndTime;
-    private Map<String, Float> regenValue = new HashMap<>();
-    private Map<String, Long> regenEndTime = new HashMap<>();
-    private float remainder;
+    public long soonestEndTime;
+    public Map<String, Float> regenValue = new HashMap<>();
+    public Map<String, Long> regenEndTime = new HashMap<>();
```

Comment:
If you can find a way, make this a map sorted on the `long` value. It will _greatly_ improve performance because at present to find min, & to remove below a specific amount then you can quickly and easily do this. It does make removing by id harder, but this is not as often needed.